appengine/datastore/src/test/java/com/example/appengine/QueriesTest.java

```diff
@@ -770,17 +770,14 @@ public void queryRestrictions_surprisingMultipleValuesTwoNotEquals_returnsMatche
     // [START surprising_behavior_example_4]
     Query q =
         new Query("Widget")
-            .setFilter(new FilterPredicate("x", FilterOperator.NOT_EQUAL, 1))
-            .setFilter(new FilterPredicate("x", FilterOperator.NOT_EQUAL, 2));
+            .setFilter(
+                CompositeFilterOperator.and(
```

Comment:
That is a strange one that seems to ignore the comment in the Datastore queries [doc](https://cloud.google.com/appengine/docs/java/datastore/queries) saying that "A query can have no more than one NOT_EQUAL filter, and a query that has one cannot have any other inequality filters". @pcostell suggests that this is because the AE Datastore **client** breaks it
to separate queries and then join the combined results.